packages/features/form-builder/FormBuilder.tsx

```diff
@@ -0,0 +1,736 @@
+          placeholder={field.placeholder}
+          label={field.label}
+          readOnly={readOnly}
+          value={value as string[]}
+          setValue={setValue as (arg: typeof value) => void}
+        />
+      </WithLabel>
+    );
+  }
+
+  if (componentConfig.propsType === "select") {
+    if (!field.options) {
+      throw new Error("Field options is not defined");
+    }
+
+    return (
+      <WithLabel field={field} readOnly={readOnly}>
+        <componentConfig.factory
+          readOnly={readOnly}
+          value={value as string}
+          placeholder={field.placeholder}
+          setValue={setValue as (arg: typeof value) => void}
+          options={field.options.map((o) => ({ ...o, title: o.label }))}
+        />
+      </WithLabel>
+    );
+  }
+
+  if (componentConfig.propsType === "multiselect") {
+    if (!field.options) {
+      throw new Error("Field options is not defined");
+    }
+    return (
+      <WithLabel field={field} readOnly={readOnly}>
+        <componentConfig.factory
+          placeholder={field.placeholder}
+          readOnly={readOnly}
+          value={value as string[]}
+          setValue={setValue as (arg: typeof value) => void}
+          options={field.options.map((o) => ({ ...o, title: o.label }))}
+        />
+      </WithLabel>
+    );
+  }
+
+  if (componentConfig.propsType === "objectiveWithInput") {
+    if (!field.options) {
+      throw new Error("Field options is not defined");
+    }
+    if (!field.optionsInputs) {
+      throw new Error("Field optionsInputs is not defined");
+    }
+    return field.options.length ? (
+      <WithLabel field={field} readOnly={readOnly}>
+        <componentConfig.factory
+          placeholder={field.placeholder}
+          readOnly={readOnly}
+          name={field.name}
+          value={value as { value: string; optionValue: string }}
+          setValue={setValue as (arg: typeof value) => void}
+          optionsInputs={field.optionsInputs}
+          options={field.options}
+        />
+      </WithLabel>
+    ) : null;
+  }
+
+  throw new Error(`Field ${field.name} does not have a valid propsType`);
+};
+
+export const FormBuilderField = ({
+  field,
+  readOnly,
+  className,
+}: {
+  field: RhfFormFields[number];
+  readOnly: boolean;
+  className: string;
+}) => {
+  const { t } = useLocale();
+  const { control, formState } = useFormContext();
+  return (
+    <div
+      data-form-builder-field-name={field.name}
+      className={classNames(className, field.hidden ? "hidden" : "")}>
+      <Controller
+        control={control}
+        // Make it a variable
+        name={`responses.${field.name}`}
+        render={({ field: { value, onChange } }) => {
+          return (
+            <div>
+              <ComponentForField
+                field={field}
+                value={value}
+                readOnly={readOnly}
+                setValue={(val: unknown) => {
+                  onChange(val);
+                }}
```

Comment:
Why not let ComponentForField accept this signature?

```suggestion
                onChange={onChange}
```